Isosurfacing_3/include/CGAL/Isosurfacing_3/internal/marching_cubes_functors.h

```diff
@@ -0,0 +1,297 @@
+//
+//     MIT License
+//
+//     Copyright (c) 2020 Roberto Grosso
+//
+//     Permission is hereby granted, free of charge, to any person obtaining a copy
+//     of this software and associated documentation files (the "Software"), to deal
+//     in the Software without restriction, including without limitation the rights
+//     to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//     copies of the Software, and to permit persons to whom the Software is
+//     furnished to do so, subject to the following conditions:
+//
+//     The above copyright notice and this permission notice shall be included in all
+//     copies or substantial portions of the Software.
+//
+//     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//     IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//     FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//     AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//     LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//     OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//     SOFTWARE.
+//
+//
+// The code below uses the version of
+// https://github.com/rogrosso/tmc available on 15th of September 2022.
+//
+
+#ifndef CGAL_ISOSURFACING_3_INTERNAL_MARCHING_CUBES_FUNCTORS_H
+#define CGAL_ISOSURFACING_3_INTERNAL_MARCHING_CUBES_FUNCTORS_H
+
+#include <CGAL/license/Isosurfacing_3.h>
+
+#include <CGAL/Isosurfacing_3/internal/tables.h>
+
+#include <CGAL/assertions.h>
+
+#ifdef CGAL_LINKED_WITH_TBB
+#include <tbb/enumerable_thread_specific.h>
+#else
+#include <vector>
+#endif
+
+#include <array>
+#include <bitset>
+
+namespace CGAL {
+namespace Isosurfacing {
+namespace internal {
+
+// Interpolate linearly between two vertex positions v0, v1 with values d0 and d1 according to the isovalue
+template <typename GeomTraits>
+typename GeomTraits::Point_3 vertex_interpolation(const typename GeomTraits::Point_3& p0,
+                                                  const typename GeomTraits::Point_3& p1,
+                                                  const typename GeomTraits::FT d0,
+                                                  const typename GeomTraits::FT d1,
+                                                  const typename GeomTraits::FT isovalue,
+                                                  const GeomTraits& gt)
+{
+  using FT = typename GeomTraits::FT;
+
+  typename GeomTraits::Compute_x_3 x_coord = gt.compute_x_3_object();
+  typename GeomTraits::Compute_y_3 y_coord = gt.compute_y_3_object();
+  typename GeomTraits::Compute_z_3 z_coord = gt.compute_z_3_object();
+  typename GeomTraits::Construct_point_3 point = gt.construct_point_3_object();
+
+  FT mu = FT(0.0);
+
+  // don't divide by 0
+  if(abs(d1 - d0) < 0.000001) // @fixme hardcoded bound
```

Comment:
Don't forget the @fixme

Reply:
Be careful not to mention people that have nothing to do with CGAL.